src/java.base/share/classes/java/lang/Process.java

```diff
@@ -153,14 +183,251 @@ public Process() {}
+     * Output to the {@code BufferedWriter} may be held in the buffer until
+     * {@linkplain BufferedWriter#flush flush} is called.
+     *
+     * @return a {@code BufferedWriter} to the standard input of the process using the charset
+     *          for the {@code native.encoding} system property
+     * @since 17
+     */
+    public final BufferedWriter outputWriter() {
+        return outputWriter(CharsetHolder.nativeCharset());
+    }
+
+    /**
+     * Returns a {@code BufferedWriter} connected to the normal input of the process
+     * using a Charset.
+     * Writes text to a character-output stream, buffering characters so as to provide
+     * for the efficient writing of single characters, arrays, and strings.
+     *
+     * <p>Characters written by the writer are encoded to bytes using {@link OutputStreamWriter}
+     * and the {@link Charset}
+     * piped into the standard input of the process represented by this {@code Process} object.
+     * Malformed-input and unmappable-character sequences are replaced with the charset's
+     * default replacement.
+     * Call the {@link BufferedWriter#flush()} method to flush buffered output to the process.
+     *
+     * <p>If the standard input of the process has been redirected using
+     * {@link ProcessBuilder#redirectInput(Redirect)
+     * ProcessBuilder.redirectInput} then this method will return a
+     * <a href="ProcessBuilder.html#redirect-input">null output writer</a>.
+     *
+     * @apiNote
+     * A {@linkplain BufferedWriter} writes characters, arrays of characters, and strings.
+     * Wrapping the {@link BufferedWriter} with a {@link PrintWriter} provides
+     * efficient buffering and formatting of primitives and objects as well as support
+     * for auto-flush on line endings.
+     * <p>
+     * Use {@link #getOutputStream} and {@link #outputWriter} with extreme care.
+     * Output to the {@code BufferedWriter} may be held in the buffer until
+     * {@linkplain BufferedWriter#flush flush} is called.
+     *
+     * @param charset the {@code Charset} to encode characters to bytes
+     * @return a {@code BufferedWriter} to the standard input of the process using the {@code charset}
+     * @throws NullPointerException if the {@code charset} is {@code null}
+     * @throws IllegalArgumentException if called more than once with different charset arguments
+     * @since 17
+     */
+    public final BufferedWriter outputWriter(Charset charset) {
+        Objects.requireNonNull(charset, "charset");
+        synchronized (this) {
+            if (outputWriter == null) {
+                outputCharset = charset;
+                outputWriter = new BufferedWriter(new OutputStreamWriter(getOutputStream(), charset));
+            } else {
+                if (!outputCharset.equals(charset))
+                    throw new IllegalArgumentException("BufferedWriter was created with charset: " + outputCharset);
```

Comment:
I'm not sure that IAE is the right exception here, I think it's closer to IllegalStateException because the first usage of  outputWriter has the side effect of setting the charset for the Process's writer. Another option here is to just put it into the "unpredictable" bucket that is using getOutputStream and outputWriter at the same time. In that case, it could just return a new BufferedWriter when it doesn't match the charset of the first usage.